### PR TITLE
Removed Swift tests config from CI pipeline

### LINKED ISSUE #4046: GitHub Actions should run unit tests only via `tuist test` and not SPM
**Describe the bug**
Currently, in the workflow `.github/workflows/tuist.yml` we run unit test twice, once via `tuist test` and again via `swift test`. 

**To Reproduce**

1. Run CI actions
2. Unit tests are run twice

**Expected behavior**
Unit tests must be run only via `tuist test`.

### PR DESCRIPTION
**Pull request by mohitsaxenaknoldus**
Resolves Closes #4046 

### Short description 📝

> Describe here the purpose of your PR.

### How to test the changes locally 🧐

> Include a set of steps for the reviewer to test the changes locally.

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [ ] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [x] In case the PR introduces changes that affect users, the documentation has been updated.
- [ ] In case the PR introduces changes that affect how the cache artifact is generated starting from the `TuistGraph.Target`, the `Constants.cacheVersion` has been updated.

### CODE REVIEW

**Review by danyf90 — APPROVED** [review]

File: projects/fourier/test/fourier/services/test/tuist/unit_test.rb [hunk 1]
```diff
@@ -17,9 +17,6 @@ def test_call
             Utilities::System
               .expects(:tuist)
               .with("test")
-            Utilities::System
-              .expects(:system)
-              .with("swift", "test")
```
Comment (ezraberch) [inline projects/fourier/test/fourier/services/test/tuist/unit_test.rb:22]: This is the internal unit test for fourier. What this does is verify that when you call `fourier test tuist unit`, fourier will call `tuist dependencies fetch`, `tuist test`, and `swift test`. Removing these 3 lines will make the unit test no longer expect `swift test` to be called, but won't actually change the fact that it is called. Thus the test will now fail.

**Review by danyf90 — CHANGES_REQUESTED** [review]
The file to be changed is `projects/fourier/lib/fourier/services/test/tuist`

File: projects/fourier/lib/fourier/services/test/tuist/unit.rb [hunk 2]
```diff
@@ -9,8 +9,6 @@ def call
             dependencies = ["dependencies", "fetch"]
             Utilities::System.tuist(*dependencies)
             Utilities::System.tuist("test")
-            Dir.chdir(Constants::TUIST_DIRECTORY) do
-              Utilities::System.system("swift", "test")
             end
```
Comment (danyf90) [inline projects/fourier/lib/fourier/services/test/tuist/unit.rb:12]: should be removed as well

File: projects/fourier/test/fourier/services/test/tuist/unit_test.rb [hunk 3]
```diff
@@ -20,7 +20,7 @@ def test_call
             Utilities::System
               .expects(:system)
               .with("swift", "test")
-
+
```
Comment (danyf90) [inline projects/fourier/test/fourier/services/test/tuist/unit_test.rb:23]: you should remove this as well.
Aslo, pay attention to not leaving empty spaces in empty line or linting will fail

**Review by danyf90 — CHANGES_REQUESTED** [review]
Could you also rebase on master? 🙏

**Comment by mohitsaxenaknoldus** [comment]
> Could you also rebase on master? pray

I don't see a master branch.

**Comment by ezraberch** [comment]
> I don't see a master branch.

`main` is the master branch.

**Comment by mohitsaxenaknoldus** [comment]
> > I don't see a master branch.
> 
> `main` is the master branch.

Up to date.

File: projects/fourier/lib/fourier/services/test/tuist/unit.rb (hunk 2 above)
Comment (danyf90) [reply projects/fourier/lib/fourier/services/test/tuist/unit.rb:12]: This doesn't seem done. I refer to the `end` of the `if`

**Comment by danyf90** [comment]
https://github.com/tuist/tuist/pull/4063#discussion_r793039468 🙏

File: projects/fourier/test/fourier/services/test/tuist/unit_test.rb [hunk 4]
```diff
@@ -17,10 +17,6 @@ def test_call
             Utilities::System
               .expects(:tuist)
               .with("test")
-            Utilities::System
-              .expects(:system)
-              .with("swift", "test")
-
             # When/Then
```
Comment (danyf90) [inline projects/fourier/test/fourier/services/test/tuist/unit_test.rb:20]:
```suggestion

            # When/Then
```

**Review by danyf90 — APPROVED** [review]

**Review by adellibovi — APPROVED** [review]

**Comment by luispadron** [comment]
@allcontributors add @mohitsaxenaknoldus for code

**Pull request merged**